js/utils/bulkCoinUpdate.js

```diff
@@ -0,0 +1,53 @@
+import $ from 'jquery';
+import { Events } from 'backbone';
+import { openSimpleMessage } from '../views/modals/SimpleMessage';
+import app from '../app';
+
+const events = {
+  ...Events,
+};
+
+export { events };
+
+let bulkCoinUpdateSave;
+
+export function isBulkCoinUpdating() {
+  return bulkCoinUpdateSave && bulkCoinUpdateSave.state() === 'pending';
+}
+
+function showError(opts) {
+  events.trigger('bulkCoinUpdateFailed');
+  const title = opts.title || app.polyglot.t('settings.storeTab.bulkListingCoinUpdate.ErrorTitle');
+  const msg = opts.msg || app.polyglot.t('settings.storeTab.bulkListingCoinUpdate.ErrorMessage');
+  openSimpleMessage(title, msg);
+}
+
+export function bulkCoinUpdate(coins) {
+  if (!Array.isArray(coins)) {
+    throw new Error('Please provide an array of accepted cryptocurrency codes.');
+  }
+
+  // dedupe the list
+  const newCoins = [...new Set(coins)];
+
+  if (!newCoins.length) {
```

Comment:
These first two shoeError calls IMHO don't make sense as user facing errors. They're really developer errors and this function should just throw them as exceptions, similar to if you're not passing in a proper array.

The user should never be able to get to this point if an update is already in progress, because the button should have a spinner.

If the user doesn't select a coin, then the view should handle it prior to calling this function whether they want to show an inline error (which is what we do in 90% of the other forms) or take the easy route (which I'm fine with) and show a simple message.